repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/ActiveMQPipelineRunner.java

```
@@ -104,10 +108,42 @@ public Boolean submitBuild(Build build) {
             LOGGER.info("Build id '" + message.getText() + ", Sent Successfully to the Queue");
             connection.close();
             return true;
-        } catch(JMSException jsme) {
-            LOGGER.info("Failed to submit build, please double check ActiveMQ server");
+        }catch (JMSException e) {
+            LOGGER.warn("Failed to submit message to ActiveMQ queue");
             return false;
         }
     }
 
+    public String receiveBuildFromQueue() {
+        try {
+            // Create a ConnectionFactory
+            ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(this.url);
+
+            // Create a Connection
+            Connection connection = connectionFactory.createConnection();
+            connection.start();
+
+            // Create a Session
+            Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+            // Create the destination (Topic or Queue)
+            Destination destination = session.createQueue(this.queueName);
+
+            // Create a MessageConsumer from the Session to the Topic or Queue
+            MessageConsumer consumer = session.createConsumer(destination);
+
+            // Wait for a message
+            Message message = consumer.receive(1000);
+
+            TextMessage textMessage = (TextMessage) message;
+            String text = textMessage.getText();
+
+            consumer.close();
+            session.close();
+            connection.close();
+            return text;
+        } catch (Exception e) {
+            return "Caught: " + e;
```

avoid silencing errors, prefer throw new RuntimeException(e)

---

Fixed.